### PR TITLE
Refactor: remove jwcrypto

### LINKED ISSUE #617: Remove dependency on `jwcrypto` and `cryptography`
This task is to make the code changes necessary for removing `jwcrypto` from our `requirements.txt` file. (`cryptography` is a transitive dependency of `jwcrypto`, so we can remove that as well.) Some of our Django models currently return `jwk.JWK` instances in some of their methods.

## Acceptance Criteria
- [ ] The `requirements.txt` file does not contain `jwcrypto` or any of its dependencies
- [ ] The Benefits app still works as expected
- [ ] Unit tests still pass

## Additional context
#141 mentions that a goal is to consolidate dependency management, which means `benefits` should not be directly depending on packages that are only needed for the Eligibility API. #596 pulls out most of the usage of `jwcrypto`, except for the ones left in `core/models.py`.

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #617 

### Reviewing this PR
 - First, review [cal-itp/eligibility-api#31](https://github.com/cal-itp/eligibility-api/pull/31)
 - Then, on this branch, make sure the `eligibility-api` package is installed by either:
    - rebuilding the `client` image and the devcontainer
    - OR running `pip install -r requirements.txt` (run `pip uninstall eligibility-api` first if you previously installed it and want to re-install it)
 - Run the Pytests at [`tests/pytest/eligibility/test_views.py`](https://github.com/cal-itp/benefits/blob/b48eaa7bca8bac2c7da2567cb67ca7f30982bbe7/tests/pytest/eligibility/test_views.py) to see that eligibility verification still works

### CODE REVIEW

**Comment by angela-tran** [comment]
@thekaveman I saw your comment on https://github.com/cal-itp/eligibility-api/pull/31. This PR will definitely change due to that, so please ignore its current state! Thanks 🙏

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,7 @@
 Authlib==1.0.1
-cryptography==37.0.2
 Django==3.2.13
 django-csp==3.7
-git+https://github.com/cal-itp/eligibility-api#egg=eligibility_api
+git+https://github.com/cal-itp/eligibility-api@a2f9932d074d10afc21713ada32eebe3841a1a3d#egg=eligibility_api
```
Comment (angela-tran) [inline requirements.txt:4]: This hash can be removed once [cal-itp/eligibility-api#31](https://github.com/cal-itp/eligibility-api/pull/31) is merged into the `main` branch over in that repo.

**Review by thekaveman — COMMENTED** [review]
This is looking a lot better!

**Review by thekaveman — APPROVED** [review]
Rebuilt both containers, ran all pytests, all passed ✅

**Pull request merged**